pages_/scan.tsx

```diff
@@ -18,92 +17,66 @@ const QrReader: ComponentType<ReaderProps> = dynamic(
   }
 );
 
-const Error = ({ error }) => <div>ERROR: {error}</div>;
-const QrTemplate = () => (
-  <div>
-    <img className="img-fluid" src={img_templateUrl}></img>
-  </div>
-);
-const Loading = () => {
-  const { t } = useTranslation();
-
-  return (
-    <div className="mt-1 d-flex align-content-center justify-content-center">
-      <Spinner />
-      <p className="ml-2 mt-1 mb-0">{t('common:loading')}</p>
-    </div>
-  );
-};
-
 const Scan = () => {
   const { t } = useTranslation();
-  const [error, setError] = React.useState(false);
-  const [processing, setProcessing] = React.useState(false);
+  const [codeError, setCodeError] = React.useState(false);
+  const [readerError, setReaderError] = React.useState(false);
   const [loaded, setLoaded] = React.useState(false);
-  const [result, setResult] = React.useState(false);
 
-  const handleError = (error) => setError(error);
+  const handleError = () => setReaderError(true);
   const handleLoad = () => setLoaded(true);
   const handleScan = (result) => {
     if (result) {
-      setProcessing(true);
-      setResult(result);
+      try {
+        const url = new URL(result);
+        if (url.origin === window.location.origin) {
```

Comment:
Are we going to use the whole URL to form the QR code? 
If we only use the shop id, is the QR code more "scannable"?